Shared/Samples/Show utility associations/ShowUtilityAssociationsView.swift

```diff
@@ -0,0 +1,263 @@
+// Copyright 2023 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct ShowUtilityAssociationsView: View {
+    /// The view model for the sample.
+    @StateObject private var model = Model()
+    
+    /// The current viewpoint of the map.
+    @State private var viewpoint: Viewpoint = .initialViewpoint
+    
+    /// The scale of the viewpoint.
+    @State private var scale: Double = .zero
+    
+    /// An image that represents an attachment symbol.
+    @State private var attachmentImage: UIImage?
+    
+    /// An image that represents a connectivity symbol.
+    @State private var connectivityImage: UIImage?
+    
+    /// The display scale of this environment.
+    @Environment(\.displayScale) private var displayScale
+    
+    var body: some View {
+        MapView(
+            map: model.map,
+            graphicsOverlays: [model.associationsOverlay]
+        )
+        .onScaleChanged {
+            scale = $0
+            Task { try await model.addAssociationGraphics(viewpoint: viewpoint, scale: scale) }
+        }
+        .onViewpointChanged(kind: .boundingGeometry) {
+            viewpoint = $0
+            Task { try await model.addAssociationGraphics(viewpoint: viewpoint, scale: scale) }
+        }
```

Comment:
These 2 modifiers are called very often - when I slightly zoom the map, there are typically 10+ calls to the `addAssociationGraphics` method. If I let go my fingers while the map pan with flicking, it calls the method many many times.

While the method is relatively low cost, it seems to introduce a visible lag. Do you think adding a boolean to stop multiple calls, or only call the method when interacting stops, is a good idea?